src/quickgui/qgsquickmapsettings.cpp

```diff
@@ -0,0 +1,214 @@
+{
+  return mMapSettings.transformContext();
+}
+
+QgsRectangle QgsQuickMapSettings::extent() const
+{
+  return mMapSettings.extent();
+}
+
+void QgsQuickMapSettings::setExtent( const QgsRectangle &extent )
+{
+  if ( mMapSettings.extent() == extent )
+    return;
+
+  mMapSettings.setExtent( extent );
+  emit extentChanged();
+}
+
+void QgsQuickMapSettings::setCenter( const QgsPoint &center )
+{
+  QgsVector delta = QgsPointXY( center ) - mMapSettings.extent().center();
+
+  QgsRectangle e = mMapSettings.extent();
+  e.setXMinimum( e.xMinimum() + delta.x() );
+  e.setXMaximum( e.xMaximum() + delta.x() );
+  e.setYMinimum( e.yMinimum() + delta.y() );
+  e.setYMaximum( e.yMaximum() + delta.y() );
+
+  setExtent( e );
+}
+
+double QgsQuickMapSettings::mapUnitsPerPixel() const
+{
+  return mMapSettings.mapUnitsPerPixel();
+}
+
+QgsRectangle QgsQuickMapSettings::visibleExtent() const
+{
+  return mMapSettings.visibleExtent();
+}
+
+QPointF QgsQuickMapSettings::coordinateToScreen( const QgsPoint &p ) const
+{
+  QgsPointXY pt( p.x(), p.y() );
+  QgsPointXY pp = mMapSettings.mapToPixel().transform( pt );
+  return pp.toQPointF();
+}
+
+QgsPoint QgsQuickMapSettings::screenToCoordinate( const QPointF &p ) const
+{
+  const QgsPointXY pp = mMapSettings.mapToPixel().toMapCoordinates( p.toPoint() );
+  return QgsPoint( pp );
+}
+
+QgsMapSettings QgsQuickMapSettings::mapSettings() const
+{
+  return mMapSettings;
+}
+
+QSize QgsQuickMapSettings::outputSize() const
+{
+  return mMapSettings.outputSize();
+}
+
+void QgsQuickMapSettings::setOutputSize( const QSize &outputSize )
+{
+  if ( mMapSettings.outputSize() == outputSize )
+    return;
+
+  mMapSettings.setOutputSize( outputSize );
+  emit outputSizeChanged();
+}
+
+double QgsQuickMapSettings::outputDpi() const
+{
+  return mMapSettings.outputDpi();
+}
+
+void QgsQuickMapSettings::setOutputDpi( double outputDpi )
+{
+  if ( qgsDoubleNear( mMapSettings.outputDpi(), outputDpi ) )
+    return;
+
+  mMapSettings.setOutputDpi( outputDpi );
+  emit outputDpiChanged();
+}
+
+QgsCoordinateReferenceSystem QgsQuickMapSettings::destinationCrs() const
+{
+  return mMapSettings.destinationCrs();
+}
+
+void QgsQuickMapSettings::setDestinationCrs( const QgsCoordinateReferenceSystem &destinationCrs )
+{
+  if ( mMapSettings.destinationCrs() == destinationCrs )
+    return;
+
+  mMapSettings.setDestinationCrs( destinationCrs );
+  emit destinationCrsChanged();
+}
+
+QList<QgsMapLayer *> QgsQuickMapSettings::layers() const
+{
+  return mMapSettings.layers();
+}
+
+void QgsQuickMapSettings::setLayers( const QList<QgsMapLayer *> &layers )
+{
+  mMapSettings.setLayers( layers );
+  emit layersChanged();
+}
+
+void QgsQuickMapSettings::onReadProject( const QDomDocument &doc )
+{
+  QDomNodeList nodes = doc.elementsByTagName( "mapcanvas" );
+  if ( nodes.count() )
+  {
+    QDomNode node = nodes.item( 0 );
+
+    mMapSettings.readXml( node );
+
+    if ( !qgsDoubleNear( mMapSettings.rotation(), 0 ) )
```

Comment:
Just wondering, is `qgsDoubleNear` also required when checking for 0? I can't see how rounding errors enter here, but I'm not sure at all.

Reply:
when compared with `==` it raises compilation warning